DEVELOPMENT.md

```diff
@@ -187,28 +144,87 @@ git remote add upstream git@github.com:tektoncd/pipeline.git
 git remote set-url --push upstream no_push
 ```
 
-_Adding the `upstream` remote sets you up nicely for regularly
-[syncing your fork](https://help.github.com/articles/syncing-a-fork/)._
+> **Note** Adding the `upstream` remote to your `.git/config` sets you up nicely for regularly [syncing your fork](https://help.github.com/articles/syncing-a-fork/) and submitting pull requests.
 
-## Kubernetes cluster
+### Configure Container Registry
 
-The recommended configuration is:
+Depending on your chosen container registry that you set in the `KO_DOCKER_REPO` environment variable, you may need to additionally configure access control to allow `ko` to authenticate to it.
 
--   Kubernetes version 1.17 or later
--   4 vCPU nodes (`n1-standard-4`)
--   Node autoscaling, up to 3 nodes
--   API scopes for cloud-platform
+#### Using Google Container Registry (GCR)
 
+If using GCR with `ko`, make sure to configure
+[authentication](https://cloud.google.com/container-registry/docs/advanced-authentication#standalone_docker_credential_helper)
+for your `KO_DOCKER_REPO` if required. To be able to push images to
+`gcr.io/<project>`, you need to run this once:
+
+```shell
+gcloud auth configure-docker
+```
+
+Besides, if your registry `KO_DOCKER_REPO` is private, then you also should create a [secret](https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets)
+and add it to `tekton-pipelines-controller` and `tekton-pipelines-webhook` serviceAccounts accordingly to pull images.
+
+1. Create a secret
+    ```yaml
+    kubectl create secret docker-registry ${SECRET_NAME} \
+    --docker-username=${USERNAME} \
+    --docker-password=${PASSWORD} \
+    --docker-email=me@here.com \
+    --namespace=tekton-pipelines
+    ```
+
+2. Add the `secret` to the K8s `ServiceAccount` configuration file
+
+    Before using `ko` to deploy `tekton-pipelines`, you need to change [the contents of the ServiceAccount](./config/200-serviceaccount.yaml) configuration file as shown below:
+
+    ```yaml
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+    name: tekton-pipelines-controller
+    namespace: tekton-pipelines
+    labels:
+        app.kubernetes.io/component: controller
+        app.kubernetes.io/instance: default
+        app.kubernetes.io/part-of: tekton-pipelines
+    imagePullSecrets:
+    - name: ${SECRET_NAME}
+    ---
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+    name: tekton-pipelines-webhook
+    namespace: tekton-pipelines
+    labels:
+        app.kubernetes.io/component: webhook
+        app.kubernetes.io/instance: default
+        app.kubernetes.io/part-of: tekton-pipelines
+    imagePullSecrets:
+    - name: ${SECRET_NAME}
+    ```
+
+---
+
+## Building and deploying
+
+### Setup a Kubernetes cluster
+
+The recommended configuration is:
+
+-  Kubernetes version 1.17 or later
+-  4 vCPU nodes (`n1-standard-4`)
+-  Node autoscaling, up to 3 nodes
```

Comment:
I'm not convinced this was ever actually needed, especially not for development.  Suggest removing unless others feel differently?

Reply:
> I'm not convinced this was ever actually needed, especially not for development. Suggest removing unless others feel differently?
> 
> Edit to add: the reason I don't think this is needed is that it introduces more complication to the management of the cluster, which I find muddies things up during development.

I think the goal here has always been to reflect min. resource requirements for effectively running/debugging/testing a complete install and ave. demand pipelines/tasks.  Those of use running on our laptops and not in some commercial cloud account truly appreciate knowing how to configure our local virtualized resources (e.g., in Docker Desktop) as we have found cases where the installed defaults truly cannot run the default installation footprint.

Reply:
> we have found cases where the installed defaults truly cannot run the default installation footprint.

This is very interesting context, thank you.  I develop primarily on a mac with Docker Desktop & Kind.  I haven't been bitten by resource problems yet but I can see how they might show up during local dev.